lib/gears/string.lua

```diff
@@ -0,0 +1,86 @@
+---------------------------------------------------------------------------
+--- String module for gears
+--
+-- @module gears.string
+---------------------------------------------------------------------------
+
+local gstring = {}
+
+local xml_entity_names = { ["'"] = "&apos;", ["\""] = "&quot;", ["<"] = "&lt;", [">"] = "&gt;", ["&"] = "&amp;" };
+--- Escape a string from XML char.
+-- Useful to set raw text in textbox.
+-- @class function
+-- @name escape
+-- @param text Text to escape.
+-- @return Escape text.
+function gstring.escape(text)
+    return text and text:gsub("['&<>\"]", xml_entity_names) or nil
+end
+
+local xml_entity_chars = { lt = "<", gt = ">", nbsp = " ", quot = "\"", apos = "'", ndash = "-", mdash = "-", amp = "&" };
+--- Unescape a string from entities.
```

Comment:
I think you should add `XML` to make it more clear that it unescape from XML entities